osu.Game/Stores/RealmArchiveModelManager.cs

```diff
@@ -132,6 +132,44 @@ public void Delete(List<TModel> items, bool silent = false)
             notification.State = ProgressNotificationState.Completed;
         }
 
+        /// <summary>
+        /// Delete videos from a list of items.
+        /// This will post notifications tracking progress.
+        /// </summary>
+        public void DeleteVideos(List<TModel> items, bool silent = false)
```

Comment:
This method cannot exist in this class because there are derived classes of `RealmArchiveModelManager` for which a "delete videos" operation makes precisely zero sense, such as:
* `ScoreModelManager`
* `SkinModelManager`

I don't even know if this method needs to exist at the model manager level at all, but even assuming it does, it should only be implemented in `BeatmapModelManager`.

Reply:
I don't think it needs to exist. Should be implemented as local as possible.

These classes mostly exist for *very* common functions, which this is not.

Reply:
The only thing that would be more local than the current setup would be to just inline the entire implementation of the operation inside `BeatmapManager` (i.e. the only call site of the added `BeatmapModelManager` method, probably ever).

Reply:
So is there anything else I would need to fix here, or is this acceptable in its current state? Do let me know, or let me know if it's just low priority to merge.

Reply:
I'm relatively fine with this method being where it is right now as it doesn't feel too global anymore and moving up to `BeatmapManager` sort of presents its own challenges in a way (`PostNotification` is not easily available there in the current state of affairs).